hera_pspec/pspecbeam.py

```diff
@@ -1,24 +1,167 @@
+            conversions.Cosmo_Conversions()
+
+    beam_freqs : array of floats
+            Frequency of beam integrals in omega_ratio in units of Hz.
+
+    omega_ratio : array of floats
+            Ratio of the integrated squared-beam power over the square of the integrated beam power
+            for each frequency in beam_freqs. i.e. Omega_pp(nu) / Omega_p(nu)^2
+
+    pspec_freqs : array of floats
+            Array of frequencies over which power spectrum is estimated in Hz.
+
+    num_steps : int, optional
+            Number of steps to use when interpolating primary beams for
+            numerical integral
+
+    stokes: str, optional
+            Which Stokes parameter's beam to compute the scalar for.
+            'pseudo_I', 'pseudo_Q', 'pseudo_U', 'pseudo_V', although currently only 'pseudo_I' is implemented
+            Default: 'pseudo_I'
+
+    taper : str, optional
+            Whether a tapering function (e.g. Blackman-Harris) is being
+            used in the power spectrum estimation.
+            Default: none
+
+    little_h : boolean, optional
+            Whether to have cosmological length units be h^-1 Mpc or Mpc
+            Value of h is obtained from conversion object stored in pspecbeam
+            Default: h^-1 Mpc
+
+    Returns
+    -------
+    scalar: float
+            [\int dnu (\Omega_PP / \Omega_P^2) ( B_PP / B_P^2 ) / (X^2 Y)]^-1
+            in h^-3 Mpc^3 or Mpc^3.
+    """
+    # get integration freqs, redshift and cosmological scalars
+    integration_freqs = np.linspace(pspec_freqs.min(), pspec_freqs.max(), num_steps, endpoint=True, dtype=np.float)
+    integration_freqs_MHz = integration_freqs / 1e6  # The interpolations are generally more stable in MHz
+    redshifts = cosmo.f2z(integration_freqs).flatten()
+    X2Y = np.array(map(lambda z: cosmo.X2Y(z, little_h=little_h), redshifts))
+
+    # Use linear interpolation to interpolate the frequency-dependent quantities
+    # derived from the beam model to the same frequency grid as the power spectrum
+    # estimation
+    beam_model_freqs_MHz = beam_freqs / 1e6
+    dOpp_over_Op2_fit = interp1d(beam_model_freqs_MHz, omega_ratio, kind='quadratic')
+    dOpp_over_Op2 = dOpp_over_Op2_fit(integration_freqs_MHz)
+
+    # Get B_pp = \int dnu taper^2 and Bp = \int dnu
+    if taper == 'none':
+        dBpp_over_BpSq = np.ones_like(integration_freqs, np.float)
+    else:
+        dBpp_over_BpSq = aipy.dsp.gen_window(len(pspec_freqs), taper)**2
+        dBpp_over_BpSq = interp1d(pspec_freqs, dBpp_over_BpSq, kind='nearest')(integration_freqs)
+    dBpp_over_BpSq /= (integration_freqs[-1] - integration_freqs[0])**2
+
+    # integrate to get scalar
+    d_inv_scalar = dBpp_over_BpSq * dOpp_over_Op2 / X2Y
+    scalar = 1.0 / integrate.trapz(d_inv_scalar, x=integration_freqs)
+
+    return scalar
 
 
 class PSpecBeamBase(object):
 
     def __init__(self, cosmo=None):
-        if cosmo != None:
+        if cosmo is not None:
```

Comment:
Just curious, what's the rationale here? I would've thought that testing for equality would be more robust than testing for identity.

Reply:
yeah I agree, however, this is just the PEP8 standard